src/main/java/calculation/CalculateCoordinatesInterpolationAzimuthLength.java

```diff
@@ -0,0 +1,92 @@
+package calculation;
+
+import model.Location;
+import model.LocationPoint;
+import org.gavaghan.geodesy.Ellipsoid;
+import org.gavaghan.geodesy.GeodeticCalculator;
+import org.gavaghan.geodesy.GlobalCoordinates;
+import utils.Utilities;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class CalculateCoordinatesInterpolationAzimuthLength {
+  public List<Location> calculatePointsOnRoutePrototype(LocationPoint locationPoint) {
+    Location location;
+    List<Location> resultsCoordinates = new ArrayList<>();
+    List<Double> newCalculatedLat = new ArrayList<>();
+    List<Double> newCalculatedLong = new ArrayList<>();
+    newCalculatedLat.add(locationPoint.getPointLatitude().get(0));
+    newCalculatedLong.add(locationPoint.getPointLongitude().get(0));
+    List<Integer> lastIndexUsed = new ArrayList<>();
+    lastIndexUsed.add(0);
+
+    double step = locationPoint.getStep();
+    int locationIndex = locationPoint.getPointDistance().size();
+    for (int oldPosition = 0; oldPosition < locationIndex; oldPosition++) {
+      double oldPositionLength = locationPoint.getPointDistance().get(oldPosition);
+      double oldPositionAzimuth = locationPoint.getPointAzimuth().get(oldPosition);
+      GeodeticCalculator geoCalc = new GeodeticCalculator();
+      Ellipsoid reference = Ellipsoid.WGS84;
+      GlobalCoordinates startPoint;
+      double extraDistance = 0;
+      if (oldPositionLength > step) {
+        while (oldPositionLength > step) {
+          int lastIndex = lastIndexUsed.get(lastIndexUsed.size() - 1);
+          double averageAngle =
+              Utilities.calculateP2PAzimuth(
+                  locationPoint.getPointLatitude().get(lastIndex),
+                  locationPoint.getPointLongitude().get(lastIndex),
+                  locationPoint.getPointLatitude().get(oldPosition),
+                  locationPoint.getPointLongitude().get(oldPosition));
+          if (oldPosition - lastIndex == 0) {
+            averageAngle = oldPositionAzimuth;
+          }
+          lastIndexUsed.add(oldPosition);
+
+          startPoint =
+              new GlobalCoordinates(
+                  newCalculatedLat.get(newCalculatedLat.size() - 1),
+                  newCalculatedLong.get(newCalculatedLong.size() - 1));
+          oldPositionLength = oldPositionLength - step;
+          double[] endBearing = new double[1];
```

Comment:
this seems to be never changed maybe can be moved to a constant? or at least moved outside the loop?